[integrations] Feat: Add Livepeer (#113)
turbo-eth/template-web3-app: integrations/livepeer/README.md

- Stream Configuration: It includes a copy-to-clipboard feature that simplifies the use of new live stream configuration information.

- Notifications: It offers toast notifications for copy-to-clipboard actions and error handling, ensuring a smooth user experience.

## API

`useCheckLivepeerApiKey`
A custom React hook that checks if the provided API Key is valid for use on Livepeer API. It includes the following properties and methods:

- `checkLivepeerApiKey(apiKey)`: A method that tries to delete a hypothetical livestream and according to its errors defines if the provided API Key is valid for Livepeer API.
  - error with status 404 `NotFoundError` means that the API key is valid and has Full API access
  - error with status 403 `PermissionError` means that the API key is valid but doesn't have Full API access
  - error with the message 'Failed to fetch' `FailedToFetchError` means that the API key is not valid, it can be an error to fetch or the valid API Key doesn't have CORS access

`useLivepeerApiKey`
A custom React hook that provides state regarding Livepeer API Key. It includes the following properties and methods:

> - `useLivepeerApiKey`: A method that provides the current state of a custom API Key set by the client 
> - `useIsLivepeerApiKeySet`: A method that checks if the Livepeer API Key is set by the client or the server and returns true if one of the cases is defined and false if neither is.
> 
> 
> ## Components

Reviewer: Fixing README formatting, otherwise the CI will break

```suggestion
- `useLivepeerApiKey`: A method that provides the current state of a custom API Key set by the client
- `useIsLivepeerApiKeySet`: A method that checks if the Livepeer API Key is set by the client or the server and returns true if one of the cases is defined and false if neither is.

## Components
```

This integration with the [Livepeer API](https://livepeer.org/) offers a user-friendly approach for managing video content and livestreams directly within the browser, thanks to its suite of reusable React Components.

## Features

- File Uploading Interface: It equips users with an interface to upload MP4 video files to an on-chain server. This feature is critical in preserving data immutability and security.

- Video Playback: It supports video playback capabilities, allowing users to watch previously uploaded videos.

- Livestream Creation: It facilitates the creation of livestreams using OBS (Open Broadcaster Software) or a web browser, offering flexible options to broadcasters.

- Livestream Playback: It enables users to watch livestreams.

- Stream Configuration: It includes a copy-to-clipboard feature that simplifies the use of new live stream configuration information.

- Notifications: It offers toast notifications for copy-to-clipboard actions and error handling, ensuring a smooth user experience.

## API

`useCheckLivepeerApiKey`
A custom React hook that checks if the provided API Key is valid for use on Livepeer API. It includes the following properties and methods:

> - `checkLivepeerApiKey(apiKey)`: A method that tries to delete a hypotetical livestream and according to it's errors definies if the provided API Key is valid for Livepeer API (errors that mean the API key is valid are 403 and 404)

Reviewer: ```suggestion
- `checkLivepeerApiKey(apiKey)`: A method that tries to delete a hypothetical livestream and according to it's errors definies if the provided API Key is valid for Livepeer API (errors that mean the API key is valid are 403 and 404)
```

`useLivepeerApiKey`
A custom React hook that provides state regarding Livepeer API Key. It includes the following properties and methods:

- `useLivepeerApiKey`: A method that provides the current state of a custom API Key set by the client 

> - `useIsLivepeerApiKeySet`: A method that checks if the Livepeer API Key is set by the client or the server and returns true if one of the cases are defined and false if netheir is.

Reviewer: ```suggestion
- `useIsLivepeerApiKeySet`: A method that checks if the Livepeer API Key is set by the client or the server and returns true if one of the cases is defined and false if neither is.
```

`FormLivepeerApiKey`
A React component that displays a form to collect the user's Livepeer API key. Once the form is submitted, the key is validated and saved if it is correct. The user is alerted if the key is invalid or if there are any other errors. If no key is set, a persistent notification prompts the user to set the key.

`FormLivepeerAsset`
A React component that renders a form to collect the user's Livepeer asset ID. Upon form submission, the component redirects the user to a page where the user can watch the video associated with the provided asset ID. The submission button is disabled until a valid asset ID is entered.

`FormLivepeerStream`
A React component that renders a form to collect the user's Livepeer stream ID. Upon form submission, the component redirects the user to a page displaying the livestream associated with the provided stream ID. The submission button is disabled until a valid stream ID is entered.

`PlayerComponent`
This is a React component that uses Livepeer's player to play a stream, file, or an IPFS URL. The type of player and the source of the content is determined by the props provided.

> If type is PlayerType.IPFS_URL, then an ipfsUrl must be provided and playbackId should not exist. In contrast, if type is PlayerType.FILE or PlayerType.STREAM, a playbackId must be provided and ipfsUrl should not exist.

Reviewer: ```suggestion
If the type is PlayerType.IPFS_URL, then an ipfsUrl must be provided and playbackId should not exist. In contrast, if the type is PlayerType.FILE or PlayerType.STREAM, a playbackId must be provided and ipfsUrl should not exist.
```
